src/app/games/lasca/tests/LascaRules.spec.ts

```diff
@@ -0,0 +1,464 @@
+            const state: LascaState = LascaState.from([
+                [__, __, __, __, __, __, __],
+                [__, _v, __, __, __, __, __],
+                [__, __, _v, __, __, __, __],
+                [__, __, __, __, __, __, __],
+                [__, __, __, __, __, __, __],
+                [__, __, __, __, __, _u, __],
+                [__, __, __, __, __, __, __],
+            ], 1).get();
+
+            // When doing so
+            const move: LascaMove = LascaMove.fromCapture([new Coord(1, 1), new Coord(3, 3)]).get();
+
+            // Then it should fail
+            const reason: string = RulesFailure.CANNOT_SELF_CAPTURE();
+            RulesUtils.expectMoveFailure(rules, state, move, reason);
+        });
+        it('should forbid backward capture with normal piece', () => {
+            // Given a board on which a normal-piece could try to capture backward
+            const state: LascaState = LascaState.from([
+                [__, __, __, __, __, __, __],
+                [__, __, __, __, __, __, __],
+                [__, __, __, __, _u, __, __],
+                [__, __, __, _v, __, __, __],
+                [__, __, __, __, __, __, __],
+                [__, __, __, __, __, __, __],
+                [__, __, __, __, __, __, __],
+            ], 1).get();
+
+            // When doing so
+            const move: LascaMove = LascaMove.fromCapture([new Coord(3, 3), new Coord(5, 1)]).get();
+
+            // Then it should fail
+            const reason: string = LascaRulesFailure.CANNOT_GO_BACKWARD();
+            RulesUtils.expectMoveFailure(rules, state, move, reason);
+        });
+        it('should allow backward capture with officer', () => {
+            // Given a board on which an officer can capture backward
+            const state: LascaState = LascaState.from([
+                [__, __, __, __, __, __, __],
+                [__, __, __, __, __, _v, __],
+                [__, __, __, __, _O, __, __],
+                [__, __, __, __, __, __, __],
+                [__, __, _v, __, __, __, __],
+                [__, __, __, __, __, __, __],
+                [__, __, __, __, __, __, __],
+            ], 2).get();
+
+            // When doing it
+            const move: LascaMove = LascaMove.fromCapture([new Coord(4, 2), new Coord(6, 0)]).get();
+
+            // Then it should be a success
+            const expectedState: LascaState = LascaState.from([
+                [__, __, __, __, __, __, Ov],
+                [__, __, __, __, __, __, __],
+                [__, __, __, __, __, __, __],
+                [__, __, __, __, __, __, __],
+                [__, __, _v, __, __, __, __],
+                [__, __, __, __, __, __, __],
+                [__, __, __, __, __, __, __],
+            ], 3).get();
+            RulesUtils.expectMoveSuccess(rules, state, move, expectedState);
+        });
+        it('should allow to do small capture when big capture available', () => {
```

Comment:
should also allow to do the big one !

Reply:
j'sais pas à quel point c'est top de mettre des titres qui sont la réponses du précédent

Reply:
ça l'est pas, chaque test devrait être autoporté par son it et les describes/beforeEach dans lesquels il se trouve, non ?

Reply:
Tout à fait, d'où mon naming actuel (erh, c'était bourré d'fautes :D )